projects/vdk-core/src/vdk/internal/core/logging.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+
+class VdkBoundLogger(logging.LoggerAdapter):
+    """
+    This class wraps around a Logger to append additional context to log records.
+    Logger-wide context can be included by providing it in the `context` parameter of the
+    `bind_logger` method.
+    Log statement-specific context can be included by including an `extra` kwarg to the log call and
+    providing a dictionary object (or any object which implements __getitem__ and __iter__) with the
+    required additional log fields.
+    It is intended to be instantiated using the `bind_logger` function and providing it with the current
+    log object and the additional context which will be included in dictionary form.
+
+    Example use:
+
+    import logging
```

Comment:
We use reStructuredText in python docs, so make sure to format it as code block so it shows up well in Help - https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks